include/envoy/http/codec.h

```diff
@@ -86,6 +93,12 @@ class StreamDecoder {
    * @param trailers supplies the decoded trailers.
    */
   virtual void decodeTrailers(HeaderMapPtr&& trailers) PURE;
+
+  /**
+   * Called with decoded METADATA.
+   * @param decoded METADATA.
+   */
+  virtual void decodeMetadata(MetadataMap& metadata_map) PURE;
```

Comment:
Do we want to pass this as a unique_ptr with move semantics so that the callee can take ownership if they desire? I suspect this is what you want, much like the other header decode APIs.

Reply:
Fixed as suggested. Thanks!